tests-legacy/Unit/Core/Filter/HashMapWhitelistFilterTest.php

```diff
@@ -105,6 +106,30 @@ public function testKeysCanBeAddedToWhitelist()
         $this->assertSame($expected, $filter->filter($subject));
     }
 
+    public function testFilteringLazyArrayReturnsNewInstanceOfSubject()
+    {
+        $subject = new TestLazyArray();
+
+        $filter = new HashMapWhitelistFilter();
+        $filter->whitelist([
+            'some_property',
+        ]);
+
+        $filteredSubject = $filter->filter($subject);
+
+        $this->assertNotSame($subject, $filteredSubject);
```

Comment:
add `assertInstanceOf` to make sure even if it is not the same object, it is the same instance :)

Reply:
You mean to check that it's still an instance of `TestLazyArray`? It may be overkill for the purpose of the test, since later I test that it still has the filtered out property.